### DISCUSSION TITLE
英文单词的自动补全

### DISCUSSION THREAD

**Discussion by hhcwyx** [ayamir/nvimdots#941] [General]
在编辑文档的时候,英文单词补全时,只能补全前面出现过的单词,对新词没有补全功能,能不能做到对新词的补全?

**Comment by charliie-dev** [comment]
https://github.com/uga-rosa/cmp-dictionary

**Reply by Jint-lzxy** [reply]
Also this looks quite similar to an issue u opened before: #414. Anything that drove you to create a new discussion?

**Reply by hhcwyx** [reply]
谢谢!我装上这个插件试试,谢谢!

**Comment by hhcwyx** [comment]
各位老师,这个插件我没有安装成功,我做了下面这些事:
1. 在plugins文件夹下创建了一个custom.lua文件,文件内容如下:
   local custom = {}
custom["uga-rosa/cmp-dictionary"] = {
	lazy = true,
	-- event = "BufRead",
	config = require("custom.cmp-dictionary"),
}
return custom
2. 在configs文件夹下创建了文件夹custom,在此文件夹下创建了文件cmp-dictionary.lua,内容如下:
   return function() -- This file MUST return a function accepting no parameter and has no return value
	require("cmp-dictionary").setup()
        require("cmp-dictionary").switcher(
            spelllang = {
           en = "/home/hhc/en.dict",
  },)
end
3. 在configs/completion/cmp.lua文件中的source部分,加上了以下内容:
    sources = {
			{ name = "nvim_lsp", max_item_count = 350 },
			{ name = "nvim_lua" },
			{ name = "luasnip" },
			{ name = "path" },
			{ name = "treesitter" },
			{ name = "spell" },
			{ name = "tmux" },
			{ name = "orgmode" },
			{ name = "buffer" },
			{ name = "latex_symbols" },
			{ name = "copilot" },
			-- { name = "codeium" },
			-- { name = "cmp_tabnine" },
			{
				name = "dictionary",
				keyword_length = 2,
			},                                                        ---这两行是加上的,我不知道是不是应该在这里加?
		},

做了以上修改,插件安装成功,但是编辑txt文件的时候,并没有出现英文单词的补全.不知道哪里做错了?

**Reply by Jint-lzxy** [reply]
https://github.com/ayamir/nvimdots/issues/414#issuecomment-1374854878

**Comment by hhcwyx** [comment]
第2点更正一下:
2.在configs文件夹下创建了文件夹custom,在此文件夹下创建了文件cmp-dictionary.lua,内容如下:
return function() -- This file MUST return a function accepting no parameter and has no return value
require("cmp-dictionary").setup()
require("cmp-dictionary").switcher({
spelllang = {
en = "/home/hhc/en.dict",
},
})
end

**Comment by hhcwyx** [comment]
谢谢!
